### ISSUE TITLE
django.core.exceptions.SynchronousOnlyOperation after detecting changes

### ISSUE THREAD

**Issue by bastbnl** [django/channels#1501]
I recently upgraded to `Django==3.0.10` from an earlier Django 3 version on Debian Linux x64 running python 3.7.6 and postgresql 12.2 using at least these packages in a virtualenv:
```
Django==3.0.10
..
channels==2.4.0
channels-redis==3.1.0
daphne==2.3.0
... 
aioredis==1.2.0
django-redis==4.12.1
hiredis==1.0.0
redis==3.2.1
... 
psycopg2==2.8.6
```

I'm using consumers that inherit from `WebsocketConsumer` and are running via `AllowedHostsOriginValidator`. I have three websocket consumers and some of them using the ORM to load objects from the database. No objects are being created, which appears to be different from the case in #1464 as there's an object being created in the consumer.

This is the full error:
```
/var/www/brownpapersession/dev/brownpapersession/brownpapersession/routing.py changed, reloading.
Traceback (most recent call last):
  File "/var/www/brwnppr/dev/env/lib/python3.7/site-packages/django/core/management/base.py", line 328, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/var/www/brwnppr/dev/env/lib/python3.7/site-packages/django/core/management/commands/runserver.py", line 60, in execute
    super().execute(*args, **options)
  File "/var/www/brwnppr/dev/env/lib/python3.7/site-packages/django/core/management/base.py", line 369, in execute
    output = self.handle(*args, **options)
  File "/var/www/brwnppr/dev/env/lib/python3.7/site-packages/channels/management/commands/runserver.py", line 59, in handle
    super().handle(*args, **options)
  File "/var/www/brwnppr/dev/env/lib/python3.7/site-packages/django/core/management/commands/runserver.py", line 95, in handle
    self.run(**options)
  File "/var/www/brwnppr/dev/env/lib/python3.7/site-packages/django/core/management/commands/runserver.py", line 102, in run
    autoreload.run_with_reloader(self.inner_run, **options)
  File "/var/www/brwnppr/dev/env/lib/python3.7/site-packages/django/utils/autoreload.py", line 599, in run_with_reloader
    start_django(reloader, main_func, *args, **kwargs)
  File "/var/www/brwnppr/dev/env/lib/python3.7/site-packages/django/utils/autoreload.py", line 584, in start_django
    reloader.run(django_main_thread)
  File "/var/www/brwnppr/dev/env/lib/python3.7/site-packages/django/utils/autoreload.py", line 299, in run
    self.run_loop()
  File "/var/www/brwnppr/dev/env/lib/python3.7/site-packages/django/utils/autoreload.py", line 305, in run_loop
    next(ticker)
  File "/var/www/brwnppr/dev/env/lib/python3.7/site-packages/django/utils/autoreload.py", line 353, in tick
    self.notify_file_changed(filepath)
  File "/var/www/brwnppr/dev/env/lib/python3.7/site-packages/django/utils/autoreload.py", line 328, in notify_file_changed
    trigger_reload(path)
  File "/var/www/brwnppr/dev/env/lib/python3.7/site-packages/django/utils/autoreload.py", line 223, in trigger_reload
    sys.exit(3)
SystemExit: 3

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/var/www/brwnppr/dev/env/lib/python3.7/site-packages/django/core/management/__init__.py", line 401, in execute_from_command_line
    utility.execute()
  File "/var/www/brwnppr/dev/env/lib/python3.7/site-packages/django/core/management/__init__.py", line 395, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/var/www/brwnppr/dev/env/lib/python3.7/site-packages/django/core/management/base.py", line 341, in run_from_argv
    connections.close_all()
  File "/var/www/brwnppr/dev/env/lib/python3.7/site-packages/django/db/utils.py", line 230, in close_all
    connection.close()
  File "/var/www/brwnppr/dev/env/lib/python3.7/site-packages/django/utils/asyncio.py", line 24, in inner
    raise SynchronousOnlyOperation(message)
django.core.exceptions.SynchronousOnlyOperation: You cannot call this from an async context - use a thread or sync_to_async.
```

It happens when I'm running the development server and it is probably unrelated to previously processed requests since the system did not process any request prior to stopping due to the exception.

**Comment by bastbnl** [comment]
The exception is not raised when I start the development server using `DJANGO_ALLOW_ASYNC_UNSAFE=1`

**Comment by bastbnl** [comment]
Seems to be caused by the [decorator](https://github.com/django/django/blob/3.0.10/django/db/backends/base/base.py#L283), which suggests it's unrelated to channels, but the error is not raised when I restart the application without channels by commenting out `channels` in `INSTALLED_APPS`

**Comment by bastbnl** [comment]
The issue is resolved after installing `daphne==2.5.0`
